### PR TITLE
change the document and analyze services to use the analysis server

### PR DESCRIPTION
**Pull request by devoncarew**
- change the document and analyze services to use the analysis server (work towards https://github.com/dart-lang/dart-services/issues/297)
- rename `bin/services.dart` to `bin/server_dev.dart` to make it clear that it's not the prod server
- upgrade to the latest `analysis_server_lib` lib version
- fix a race condition with getFixes, completions, and document (found using the fuzz tester)
- update the benchmarks and fuzz tester to use the analysis server based services

@lukechurch, @danrubel - things are looking good - dartpad (client and server) work locally, tests pass, and the fuzz tester seems happy.

### CODE REVIEW

File: lib/src/analysis_server.dart [hunk 1]
```diff
@@ -37,11 +39,16 @@ class AnalysisServerWrapper {
   /// Instance to handle communication with the server.
   AnalysisServer analysisServer;
 
-  AnalysisServerWrapper(this.sdkPath) {
+  AnalysisServerWrapper(this.sdkPath, bool useStrongMode) {
     _logger.info("AnalysisServerWrapper ctor");
```
Comment (lukechurch) [inline lib/src/analysis_server.dart]: Could you stick the value of useStrongMode in the logging call?

_logger.info("AnalysisServerWrapper ctor, strong mode: $useStrongMode");

File: tool/fuzz_driver.dart [hunk 2]
```diff
@@ -135,7 +135,7 @@ setupTools(String sdkPath) async {
   apiServer = new ApiServer(apiPrefix: '/api', prettyPrint: true)
     ..addApi(server);
 
-  analysisServer = new analysis_server.AnalysisServerWrapper(sdkPath);
+  analysisServer = new analysis_server.AnalysisServerWrapper(sdkPath, true);
```
Comment (lukechurch) [inline tool/fuzz_driver.dart]: We need to drive both the strong and non-Strong analysis server with these calls. I can fix this up, could you add a TODO for me?

File: tool/fuzz_driver.dart [hunk 3]
```diff
@@ -432,8 +431,14 @@ enum OperationType {
...
-log(dynamic str) {
+final int termWidth = io.stdout.hasTerminal ? io.stdout.terminalColumns : 200;
+
+void log(dynamic obj) {
   if (_VERBOSE) {
-    print("${new DateTime.now()} $str");
+    String str = '${new DateTime.now()} $obj';
+    if (str.length > termWidth) {
+      str = str.substring(0, termWidth - 4) + '...';
+    }
```
Comment (lukechurch) [inline tool/fuzz_driver.dart]: Let's not do this in the case of fuzz_driver, we parse the results of these files with support tools for downstream analysis.

Comment (lukechurch) [inline tool/fuzz_driver.dart]: If you wish, we can change that in a subsequent PR, but for now, let's just output the whole thing.

Comment (devoncarew) [inline tool/fuzz_driver.dart]: sure, reverted

File: tool/fuzz_driver.dart (hunk 2 above)
Comment (devoncarew) [inline tool/fuzz_driver.dart]: makes sense; I'll add a todo

File: lib/src/analysis_server.dart [hunk 4]
```diff
@@ -37,11 +39,16 @@ class AnalysisServerWrapper {
   /// Instance to handle communication with the server.
   AnalysisServer analysisServer;
 
-  AnalysisServerWrapper(this.sdkPath) {
+  AnalysisServerWrapper(this.sdkPath, bool useStrongMode) {
```
Comment (devoncarew) [inline lib/src/analysis_server.dart]: done

**Pull request merged**